### DISCUSSION TITLE
Avoid list or dictionary as function input

### DISCUSSION THREAD

**Discussion by Nanco-L** [MDIL-SNU/SIMPLE-NN_v2#7] [General]
https://github.com/MDIL-SNU/SIMPLE-NN_v2/blob/03aa41d1a7a226b0708516af640f6ad69540efed/simple_nn_v2/features/symmetry_function/__init__.py#L248

https://github.com/MDIL-SNU/SIMPLE-NN_v2/blob/03aa41d1a7a226b0708516af640f6ad69540efed/simple_nn_v2/features/symmetry_function/__init__.py#L275

In the above lines, `result` is the dictionary that contains symmetry function values, their derivatives, and other values we need to know for training a neural network.
But in Python, the change of list or dictionary in the function is not discarded after the function call ends.
For example,
```Python
def in_function_change(mylist):
    mylist.append(3)

aa = [1, 2]
in_function_change(aa)
print(aa)
# >> [1, 2, 3]
```
Thus, basically, you do not need to use a list or dictionary as a function input.

In addition, I recommend changing above functions like below
```Python
def _set_EFS(self, snapshot):
    if not (self.inputs['refdata_format']=='vasp' or self.inputs['refdata_format']=='vasp-xdatcar'):
        if ase.__version__ >= '3.18.0':
            E = snapshot.get_potential_energy(force_consistent=True)
        else:
            E = snapshot.get_total_energy()
        try:
            F = snapshot.get_forces()
        except:
            if self.parent.inputs['neural_network']['use_force']:
                err = "There is not force information! Set 'use_force' = false"
                #if comm.rank == 0:
                self.parent.logfile.write("\nError: {:}\n".format(err))
                raise NotImplementedError(err)
        try:
            S = -snapshot.get_stress()/units.GPa*10
            # ASE returns the stress tensor by voigt order xx yy zz yz zx xy
            S = S[[0, 1, 2, 5, 3, 4]]
        except:
            if self.parent.inputs['neural_network']['use_stress']:
                err = "There is not stress information! Set 'use_stress' = false"
                #if comm.rank == 0:
                self.parent.logfile.write("\nError: {:}\n".format(err))
                raise NotImplementedError(err)
    return E, F, S
```

I think this form can make the test code simpler.

**Comment by JisuJung928** [comment]
I totally agree with @Nanco-L `s suggestion.
